### ISSUE TITLE
eval and environment

### ISSUE THREAD

**Issue by shakna-israel** [gambit/gambit#204]
Are environments in Gambit up to the R5RS spec?

For example:

```
$ (eval (read) (null-environment))
$ (+ 1 1)
> 2
```

Which is fairly unexpected.

Also:

```
$ (scheme-report-environment 9)
> 9
$ (scheme-report-environment "What")
> "What"
```

Why is an environment acting like eval? Not just supplying bindings?

**Comment by feeley** [comment]
Fixed by commit 09284455ae7c7e8dff4a71b8a60e5315e04fbe61

**Comment by shakna-israel** [comment]
Nicely done!
